clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -1368,7 +1368,7 @@ private int partition(ProducerRecord<K, V> record, byte[] serializedKey, byte[]
         if (record.partition() != null)
             return record.partition();
 
-        if (partitioner != null) {
+        if (partitioner != null && partitioner.partitionsTopic(record.topic())) {
```

Comment:
there's a slight change to the behavior of the current API, could you modify the documentation?